src/modules/Search/Search.js

```diff
@@ -27,6 +27,10 @@ import SearchResults from './SearchResults'
 
 const debug = makeDebugger('search')
 
+const overrideSearchInputProps = (inputProps) => ({
+  input: { className: 'prompt', tabIndex: '0', autoComplete: 'off', ...inputProps.input },
+})
```

Comment:
Shame on us, but we don't have deep handling of shorthands. This an edge case, actually.
It should be a bit more smart as we accept more things to shorthands that just an object:

```jsx
const overrideSearchInputProps = (predefinedProps) => ({
  const { input } = predefinedProps

  if (_.isUndefined(input)) return { className: 'prompt' }
  if (_.isPlainObject(input)) return { ...input, className: cx(input.className, 'prompt') }

  return input
}
```

- `isUndefined` will handle default case
- `isPlainObject` will merge `className` properly as in our implementation you will loose `prompt`

***

Let's also move it to class method:

```js
handleSearchInputOverrides = (predefinedProps) => ({
```

Reply:
You're right. Since the functional form of `overrideProps` always provides an object, I accidentally conflated it by assuming that the nested `.input` property would be an object.

However with the suggested changes it now feels too hacky; factories.js already has logic for merging classnames (as well as the `style` prop) and determining which type of shorthand it is. Therefore I think it probably needs another approach.